### PR TITLE
Introduce tests for cache attach/detach feature

### PR DESCRIPTION
**Pull request by katlapinka**

### CODE REVIEW

File: test/functional/tests/cache_ops/test_attach_detach.py [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+    with TestRun.step("Detach first cache device"):
+        cache1.detach()
+
+    with TestRun.step("Try to attach the other cache device to first cache without force flag"):
+        try:
+            cache1.attach(device=cache_dev2)
+            TestRun.fail("Cache attached successfully"
+                         "Expected: cache fail to attach")
+        except CmdException:
+            TestRun.LOGGER.info("Cache attach failed as expected")
```
Comment (Deixx) [inline test/functional/tests/cache_ops/test_attach_detach.py:107]: maybe check if proper message is displayed?

File: test/functional/tests/cache_ops/test_attach_detach.py [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+        cache_dev.create_partitions([Size(2, Unit.GibiByte)])
+        cache_dev = cache_dev.partitions[0]
+
+        cache_dev2 = TestRun.disks["cache2"]
+        cache_dev2.create_partitions([Size(3, Unit.GibiByte)])
+        cache_dev2 = cache_dev2.partitions[0]
+
+        core_dev = TestRun.disks["core"]
+        core_dev.create_partitions([Size(6, Unit.GibiByte)])
+        core_dev = core_dev.partitions[0]
```
Comment (Deixx) [inline test/functional/tests/cache_ops/test_attach_detach.py:161]: prepare devices in a separate step

File: test/functional/tests/cache_ops/test_attach_detach.py [hunk 3]
```diff
@@ -0,0 +1,258 @@
...
+
+
+@pytest.mark.require_disk("cache", DiskTypeSet([DiskType.optane, DiskType.nand]))
+@pytest.mark.require_disk("core", DiskTypeLowerThan("cache"))
+@pytest.mark.parametrizex("cache_mode", CacheMode)
+def test_stop_cache_during_attach(cache_mode):
+    """
+    title: Test cache stop during attach.
+    description: |
+        Test the ability of OpenCAS to handle concurrent attach and cache stop.
```
Comment (Deixx) [inline test/functional/tests/cache_ops/test_attach_detach.py:209]: I'm not a fan of putting 'OpenCAS' in descriptions - we know what we're testing.
'Test for handling concurrent cache attach and stop' is sufficient.

File: test/functional/tests/fault_injection/test_fault_injection_interrupts.py [hunk 4]
```diff
@@ -487,6 +495,149 @@ def test_interrupt_cache_stop(cache_mode, filesystem):
...
 
 
+@pytest.mark.require_disk("cache", DiskTypeSet([DiskType.nand, DiskType.optane]))
+@pytest.mark.require_disk("core", DiskTypeLowerThan("cache"))
+@pytest.mark.parametrizex("cache_mode", CacheMode)
+def test_interrupt_attach(cache_mode):
+    """
+    title: Test for attach interruption.
+    description: |
+        Validate if CAS correctly handles interruption of attach.
```
Comment (Deixx) [inline test/functional/tests/fault_injection/test_fault_injection_interrupts.py:505]: CAS :/

File: test/functional/tests/fault_injection/test_fault_injection_interrupts.py [hunk 5]
```diff
@@ -487,6 +495,149 @@ def test_interrupt_cache_stop(cache_mode, filesystem):
...
+@pytest.mark.require_disk("cache", DiskTypeSet([DiskType.optane, DiskType.nand]))
+@pytest.mark.require_disk("core", DiskTypeLowerThan("cache"))
+def test_detach_interrupt_cache_flush(filesystem, cache_mode):
+    """
+    title: Test for flush interruption using cache detach operation.
+    description: |
+        Validate if CAS correctly handles detach during cache flush.
+    pass_criteria:
+      - No system crash.
+      - CAS detach operation doesn't stop cache flush.
```
Comment (Deixx) [inline test/functional/tests/fault_injection/test_fault_injection_interrupts.py:570]: CAS

File: test/functional/tests/fault_injection/test_fault_injection_interrupts.py [hunk 6]
```diff
@@ -487,6 +495,149 @@ def test_interrupt_cache_stop(cache_mode, filesystem):
...
+        )
+        fio_pid = fio.run()
+
+        if cache.get_dirty_blocks() <= Size.zero():
+            TestRun.fail("Failed to populate cache with dirty data")
+        if core.get_dirty_blocks() <= Size.zero():
+            TestRun.fail("There is no dirty data on core")
+
+    with TestRun.step("Wait for fio to end"):
+        TestRun.executor.wait_cmd_finish(fio_pid)
```
Comment (Deixx) [inline test/functional/tests/fault_injection/test_fault_injection_interrupts.py:614]: do we need this?

File: test/functional/tests/io/test_attach_detach_during_io.py [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+@pytest.mark.require_disk("core", DiskTypeLowerThan("cache"))
+@pytest.mark.parametrizex("cache_mode", CacheMode)
+def test_attach_detach_during_io(cache_mode):
+    """
+    title: Test for attach/detach cache during I/O.
+    description: |
+        Validate if CAS attach detach operation doesn't interrupt
+        I/O on exported object
+    pass_criteria:
+      - CAS doesn't crash during attach and detach.
```
Comment (Deixx) [inline test/functional/tests/io/test_attach_detach_during_io.py:39]: CAS

File: test/functional/tests/memory/test_detach_memory_release.py [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+
+@pytest.mark.require_disk("cache", DiskTypeSet([DiskType.nand, DiskType.optane]))
+@pytest.mark.require_disk("core", DiskTypeLowerThan("cache"))
+def test_detach_memory_release():
+    """
+    title: Test for CAS to detect if memory was released after detach operation.
+    description: |
+        Validate if ctx was released after detach operation.
+    pass_criteria:
+      - CAS releases memory used by cache device after detach operation.
```
Comment (Deixx) [inline test/functional/tests/memory/test_detach_memory_release.py:27]: CAS

**Review by Deixx — APPROVED** [review]

**Review by Kamoppl — APPROVED** [review]
